src/cli/split-docs/index.ts

```diff
@@ -0,0 +1,377 @@
+      node.discriminator.mapping &&
+      isObject(node.discriminator.mapping)
+    ) {
+      const { mapping } = node.discriminator;
+      for (const name of Object.keys(mapping)) {
+        if (isString(mapping[name]) && isStartsWithComponents(mapping[name])) {
+          replace(node.discriminator.mapping, name);
+        }
+      }
+    }
+  });
+
+  function replace(node: refObj, key: string) {
+    const splittedNode = node[key].split('/');
+    const name = splittedNode.pop();
+    const groupName = splittedNode[2];
+    const filesGroupName = componentFiles[groupName];
+    if (!filesGroupName || !filesGroupName[name!]) return;
+    let filename = path.relative(relativeFrom, filesGroupName[name!].filename);
+    if (!filename.startsWith('.')) { filename = '.' + path.sep + filename; }
+    node[key] = filename;
+  }
+}
+
+function implicitlyReferenceDiscriminator(
+  obj: any,
+  defName: string,
+  filename: string,
+  schemaFiles: any
+) {
+  if (!obj.discriminator) return;
+
+  const defPtr = `#/${COMPONENTS}/${OPENAPI3_COMPONENT.Schemas}/${defName}`;
+  const implicitMapping = {} as any;
+  for (const [name, { inherits, filename: parentFilename }] of Object.entries(schemaFiles) as any) {
+    if (inherits.indexOf(defPtr) > -1) {
+      const res = path.relative(path.dirname(filename), parentFilename);
+      implicitMapping[name] = res.startsWith('.') ? res : '.' + path.sep + res;
+    }
+  }
+
+  if (isNotObjectKeys(implicitMapping)) return;
+  const discriminatorPropSchema = obj.properties[obj.discriminator.propertyName];
+  const discriminatorEnum = discriminatorPropSchema && discriminatorPropSchema.enum;
+
+  const mapping = (obj.discriminator.mapping = obj.discriminator.mapping || {});
+  for (const name of Object.keys(implicitMapping)) {
+    if (discriminatorEnum && !discriminatorEnum.includes(name)) { continue; }
+    if (mapping[name] && mapping[name] !== implicitMapping[name]) {
+      process.stderr.write(yellow(
+        `warning: explicit mapping overlaps with local mapping entry ${red(name)} at ${blue(filename)}, Check manually, please`
+      ));
+    }
+    mapping[name] = implicitMapping[name];
+  }
+}
+
+function isNotSecurityComponentType(componentType: string) {
+  return componentType !== OPENAPI3_COMPONENT.SecuritySchemes
+}
+
+function findComponentTypes(components: any) {
+  return OPENAPI3_COMPONENTS
+    .filter(item => isNotSecurityComponentType(item) && Object.keys(components).includes(item))
+    .map(type => ({ name: type, data: components[type] }))
+}
+
+function isFileNotEqual(filename: string, componentData: string, componentName: string) {
+  const result = fs.existsSync(filename) && !isEqual(readYaml(filename), componentData);
+  if (result) {
+    process.stderr.write(yellow(
+      `warning: conflict for ${componentName} - file already exists
+         with different content: ${blue(filename)} ... Skip.\n`
+    ));
+  }
+  return result;
+}
+
+function cleanupComponentsExceptSecuritySchemes(
```

Comment:
Every function with “cleanup” is named slightly incorrect. The function is a verb: `goToSleep()` or `eatBreakfast()`.  In this case “cleanup” itself is a noun, not to be confused with “clean up” which would have “clean” be the verb. So the “U” should be capitalized for camel casing.  

Perhaps a better verb would be “remove”. It avoids the “up”. “Remove empty properties” “remove components except security schemes”, etc..